src/spatialexperiment/_imgutils.py

```diff
@@ -1,12 +1,87 @@
 from typing import Union
 
+import os
+from io import BytesIO
+import requests
+from PIL import Image
 from biocframe import BiocFrame
+from .SpatialImage import construct_spatial_image_class
 
 __author__ = "keviny2"
 __copyright__ = "keviny2"
 __license__ = "MIT"
 
 
+
+def read_image(input_image):
+    """Read image from PIL Image, file path, or URL.
+
+    Args:
+        input_image: PIL Image, string path to local file, or URL string.
+
+    Returns:
+        The loaded image.
+
+    Raises:
+        TypeError: If input is not PIL Image, path string, or URL string.
+    """
+    if isinstance(input_image, Image.Image):
+        return input_image
+    
+    if isinstance(input_image, str):
+        if input_image.startswith(('http://', 'https://')):
```

Comment:
slightly more reliable approach is to do https://github.com/BiocPy/SpatialExperiment/blob/main/src/spatialexperiment/SpatialImage.py#L664